### DISCUSSION TITLE
Xuanwo's VISION: Data Freedom

### DISCUSSION THREAD

**Discussion by giscus** [Xuanwo/blog#425] [Announcements]
# Xuanwo's VISION: Data Freedom

An infrastructure engineer, focused on distributed storage system

https://xuanwo.io/2024/04-xuanwo-vision/

**Comment by xxchan** [comment]
It’s quite an ambitious vision! But I don’t see you talked about the WHY part, which might be also very interesting (and perhaps can attract more people to join you) :p 
First of all, not all (end) user cares about “data freedom”. Then compared with other things like “local-first”, why do you want to achieve more things like “in any language”? This seems further from end users. Anyway, why “data freedom” matters to you?

Regarding the WHAT part, what’s the position of RBIR in the vision? OpenDAL is obviously related, but why “bit data”?

**Reply by Xuanwo** [reply]
First of all, thanks a lot for reading my post and leave your feedback!

> But I don’t see you talked about the WHY part, which might be also very interesting (and perhaps can attract more people to join you) :p

Great catch! This post originally included a "WHY" section, but I decided to remove it before publishing to better organize my thoughts. It will cover a lengthy history of my childhood, my first coding experience, and my first job, among other things. If you're interested, I propose writing a dedicated post about this topic when I'm ready.

> First of all, not all (end) user cares about “data freedom”. Then compared with other things like “local-first”, why do you want to achieve more things like “in any language”? This seems further from end users. 

Apologies for the confusing use of the term `user`. Although it appears multiple times, its meaning varies depending on the context. At times, I also refer to developers as users. 

While not all users care about with `data freedom`, it is true that all will be impacted by it. `In any language` allows developers to access data using their preferred or most familiar programming language, enabling them to build applications more efficiently.

> Regarding the WHAT part, what’s the position of RBIR in the vision? OpenDAL is obviously related, but why “bit data”?

Another good question. 

As I mentioned in the "How" section, I advance my vision through `RBIR`. The reason is simple: I've found a company, @datafuselabs, that aligns closely with aspects of my vision. It's the best starting point for me. By the way, my work experience at Databend greatly contributed to solidifying my vision.

If I discover another company in the multiverse that focuses on developing note-taking software, I might start with a `local-first` approach. :rofl: 

So the point here is: I've chosen `RBIR` as the current method to advance my vision. It's not part of my VISION, it's a tool, a method. If the context changes and I find a better approach, I will adapt accordingly.

**Reply by cnsky1103** [reply]
suprised to find out that you are the founder of datafuselabs! After reading this reply I still have some confusions: won't the vision turns out to be another super-cool ETL system, or even just another navicat? 'cause as far as I know, accessing data is really just the beginning of solutions to real-world problems, even if we can access data freely, how will data flow?
Forgive me if I'm understanding Data Freedom wrong, since I have not dived into this area for long time, but it will be nice if you can describe your vision's final output!

Best regards

**Comment by tianheg** [comment]
I'd love to join you on your journey, but I'll be running for my life for the foreseeable future and have less time to participate, but I'll do my best to contribute!

**Reply by Xuanwo** [reply]
Greatly appreciated! Prioritize life first, then chase your vision. Welcome to join the journey.

**Comment by jeffcai** [comment]
It rocks!

**Comment by bjm1244** [comment]
I'm telling you because there is no pola-rs among RBIR projects!
https://pola.rs/
I deal with Rust at a beginner level, but I want to participate in the RBIR project.
